### ISSUE TITLE
zipkin-lens tag value doesn't wrap

### ISSUE THREAD

**Issue by drolando** [openzipkin/zipkin#2416]
In zipkin lens if your tag has a very long value which is larger than the window the string doesn't wrap (it does in the old ui) and it just gets cut off.

**Comment by drolando** [comment]
cc @tacigar

**Comment by codefromthecrypt** [comment]
Here is the relevant code from the old ui

spanPanel.js
```javascript

function escapeHtml(string) {
  return String(string).replace(/[&<>"'`=\/]/g, s => entityMap[s]);
}

// Normal values are formatted in traceToMustache. However, Quoted json values
// end up becoming javascript objects later. For this reason, we have to guard
// and stringify as necessary.

// annotations are named events which shouldn't hold json. If someone passed
// json, format as a single line. That way the rows corresponding to timestamps
// aren't disrupted.
export function formatAnnotationValue(value) {
  const type = $.type(value);
  if (type === 'object' || type === 'array' || value == null) {
    return escapeHtml(JSON.stringify(value));
  } else {
    return escapeHtml(value.toString()); // prevents false from coercing to empty!
  }
}

// Tags sometimes have large values, for example json representing a query or a
// stack trace. Format these so that they don't scroll off the side of the screen.
export function formatTagValue(value) {
  const type = $.type(value);
  if (type === 'object' || type === 'array' || value == null) {
    return `<pre><code>${escapeHtml(JSON.stringify(value, null, 2))}</code></pre>`;
  }
  const result = value.toString();
  // Preformat if the text includes newlines
  return result.indexOf('\n') === -1 ? escapeHtml(result)
    : `<pre><code>${escapeHtml(result)}</code></pre>`;
}
```

tests
```javascript

describe('formatAnnotationValue', () => {
  it('should return same value when string', () => {
    formatAnnotationValue('foo').should.equal('foo');
  });

  it('should return string when false', () => {
    formatAnnotationValue(false).should.equal('false');
  });

  it('should return string when true', () => {
    formatAnnotationValue(true).should.equal('true');
  });

  it('should format object as one-line json', () => {
    formatAnnotationValue({foo: 'bar'}).should.equal(
      '{&quot;foo&quot;:&quot;bar&quot;}'
    );
  });

  it('should format array as one-line json', () => {
    formatAnnotationValue([{foo: 'bar'}, {baz: 'qux'}]).should.equal(
      '[{&quot;foo&quot;:&quot;bar&quot;},{&quot;baz&quot;:&quot;qux&quot;}]'
    );
  });

  it('should format null as json', () => {
    formatAnnotationValue(null).should.equal('null');
  });

  it('should escape html', () => {
    formatAnnotationValue('<script>alert(1)</script>').should.equal(
      '&lt;script&gt;alert(1)&lt;&#x2F;script&gt;'
    );
  });
});

describe('formatTagValue', () => {
  it('should return same value when string', () => {
    formatTagValue('foo').should.equal('foo');
  });

  it('should return string when false', () => {
    formatTagValue(false).should.equal('false');
  });

  it('should return string when true', () => {
    formatTagValue(true).should.equal('true');
  });

  it('should format object as pre-formatted multi-line json', () => {
    formatTagValue({foo: 'bar'}).should.equal(
      '<pre><code>{\n  &quot;foo&quot;: &quot;bar&quot;\n}</code></pre>'
    );
  });

  it('should format array as pre-formatted multi-line json', () => {
    formatTagValue([{foo: 'bar'}, {baz: 'qux'}]).should.equal(
      '<pre><code>[\n  {\n    &quot;foo&quot;: &quot;bar&quot;\n  },\n'
      + '  {\n    &quot;baz&quot;: &quot;qux&quot;\n  }\n]</code></pre>'
    );
  });

  it('should format null as pre-formatted json', () => {
    formatTagValue(null).should.equal('<pre><code>null</code></pre>');
  });

  it('should format multi-line string as pre-formatted', () => {
    formatTagValue('foo\nbar\n').should.equal(
      '<pre><code>foo\nbar\n</code></pre>'
    );
  });

  it('should escape html', () => {
    formatTagValue('<script>alert(1)</script>').should.equal(
      '&lt;script&gt;alert(1)&lt;&#x2F;script&gt;'
    );
  });
});
```

**Comment by codefromthecrypt** [comment]
ps @drolando noted offline that probably this could be done in a different way (ex with css). I didn't intend to recommend above, just noting for background

**Comment by drolando** [comment]
Fixed by #2434
